util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -165,4 +171,56 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
           String.format("Invalid proxy address '%s'.", proxyAddress), e);
     }
   }
+
+  static class KeepAliveSslSocketFactory extends SSLSocketFactory {
```

Comment:
May we add at least some kind of test for this functionality?

Reply:
Added unit tests.